collectors/pod_test.go

```diff
@@ -468,25 +476,29 @@ func TestPodCollector(t *testing.T) {
 								Name: "pod2_con1",
 								Resources: v1.ResourceRequirements{
 									Requests: map[v1.ResourceName]resource.Quantity{
-										v1.ResourceCPU:    resource.MustParse("400m"),
-										v1.ResourceMemory: resource.MustParse("300M"),
+										v1.ResourceCPU:       resource.MustParse("400m"),
+										v1.ResourceMemory:    resource.MustParse("300M"),
+										v1.ResourceNvidiaGPU: resource.MustParse("3"),
 									},
 									Limits: map[v1.ResourceName]resource.Quantity{
-										v1.ResourceCPU:    resource.MustParse("400m"),
-										v1.ResourceMemory: resource.MustParse("300M"),
+										v1.ResourceCPU:       resource.MustParse("400m"),
+										v1.ResourceMemory:    resource.MustParse("300M"),
+										v1.ResourceNvidiaGPU: resource.MustParse("3"),
 									},
 								},
 							},
 							v1.Container{
 								Name: "pod2_con2",
 								Resources: v1.ResourceRequirements{
 									Requests: map[v1.ResourceName]resource.Quantity{
-										v1.ResourceCPU:    resource.MustParse("500m"),
-										v1.ResourceMemory: resource.MustParse("400M"),
+										v1.ResourceCPU:       resource.MustParse("500m"),
+										v1.ResourceMemory:    resource.MustParse("400M"),
+										v1.ResourceNvidiaGPU: resource.MustParse("5"),
 									},
 									Limits: map[v1.ResourceName]resource.Quantity{
-										v1.ResourceCPU:    resource.MustParse("500m"),
-										v1.ResourceMemory: resource.MustParse("400M"),
+										v1.ResourceCPU:       resource.MustParse("500m"),
+										v1.ResourceMemory:    resource.MustParse("400M"),
+										v1.ResourceNvidiaGPU: resource.MustParse("5"),
```

Comment:
could/should this also be non-integer values? if so, can you mix things up a bit so the tests also cover that?